### PR TITLE
Add test coverage with stubtest, flake8, black, and isort

### PR DESCRIPTION
**Pull request by ShaneHarvey**
This PR adds test coverage inspired by [typeshed](https://github.com/python/typeshed). The test installs pymongo and pymongo-stubs and runs stubtest on the pyi files. Stubtest is a tool (from mypy) which compares the type annotations in the stubs to the actual objects at runtime. For example this error means that the stub file contains a method that the actually module does not have:
```
$ stubtest pymongo
error: pymongo.client_session.ClientSession.__copy__ is not present at runtime
Stub: at line 101
def (self: pymongo.client_session.ClientSession)
Runtime:
MISSING
```

This PR also adds flake8, black, and isort for automatic formatting.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#       E305 expected 2 blank lines
+#       F401 imported but unused
+#       E701 multiple statements on one line (colon)
+#       F821 undefined name
+
+[flake8]
+per-file-ignores =
+  *.pyi: E301, E302, E305, F401, E701, F821
+
+exclude = .venv*,@*,.git
```
Comment (prashantmital) [inline .flake8:13]: What is `@*`?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          flake8
+
+  black:
+    name: Check formatting with black
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: |
+          pip install $(grep black requirements-tests-py3.txt)
```
Comment (prashantmital) [inline .github/workflows/tests.yml:25]: I've never seen this methodology for installing testing requirements for different setups from a single file - interesting. Any particular reason for doing it this way?

File: tests/stubtest_whitelist/common.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+pymongo.client_session.ClientSession.__copy__
```
Comment (prashantmital) [inline tests/stubtest_whitelist/common.txt:1]: I am guessing 'whitelisting' this means we are ignoring some stubtest errors that arise from it? Can you explain what the error(s) is/are?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[tool.black]
```
Comment (prashantmital) [inline pyproject.toml:1]: Is the use of `poetry` enforced by GitHub actions? Is this preinstalled on instances spawned by the testbed?

File: bson-stubs/py3compat.pyi [hunk 5]
```diff
@@ -1,28 +0,0 @@
-from collections import abc as abc
```
Comment (prashantmital) [inline bson-stubs/py3compat.pyi:1]: 🎉

File: .flake8 (hunk 1 above)
Comment (ShaneHarvey) [reply .flake8:13]: Not sure, I copied this config from [typeshed](https://github.com/python/typeshed/blob/master/.flake8). We can probably remove it.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (ShaneHarvey) [reply .github/workflows/tests.yml:25]: I copied this idea from [typeshed](https://github.com/python/typeshed/blob/master/.github/workflows/tests.yml). The idea is that this job only needs to install the dependencies for black, we don't need everything in `requirements-tests-py3.txt`.

File: pyproject.toml (hunk 4 above)
Comment (ShaneHarvey) [reply pyproject.toml:1]: We're not using `poetry`. This file configures black and isort.

File: tests/stubtest_whitelist/common.txt (hunk 3 above)
Comment (ShaneHarvey) [reply tests/stubtest_whitelist/common.txt:1]: Yep the error is in the PR description:

> For example this error means that the stub file contains a method that the actually module does not have:
```
$ stubtest pymongo
error: pymongo.client_session.ClientSession.__copy__ is not present at runtime
Stub: at line 101
def (self: pymongo.client_session.ClientSession)
Runtime:
MISSING
```

This is caused by the ClientSession.__copy__ API added in https://jira.mongodb.org/browse/PYTHON-1915.

File: pyproject.toml (hunk 4 above)
Comment (prashantmital) [reply pyproject.toml:1]: What tool reads this file?

**Review by prashantmital — APPROVED** [review]

File: pyproject.toml (hunk 4 above)
Comment (ShaneHarvey) [reply pyproject.toml:1]: black and isort.

**Pull request merged**